File: 1758-minimum-changes-to-make-alternating-binary-string/1758-minimum-changes-to-make-alternating-binary-string.cpp
class Solution {
public:
    int minOperations(string s) {
        /*
     int count=0;

     for(int i=0;i<s.length();i++){
         if(s[i]-'0'==i%2)
         count++;
     }

     return min(count,(int)s.length()-count);
     */
     int count=0;
        
        for(int i=1;i<s.length();i++){
            if(s[i-1]==s[i]){
                if(s[i-1]=='0'){
                    s[i]='1';
                    count++;
                }
                else if(s[i-1]=='1'){
                    s[i]='0';
                    count++;
                }
            }
        }
        return min(count,(int)s.length()-count);
    }
};

File: 1758-minimum-changes-to-make-alternating-binary-string/1758-minimum-changes-to-make-alternating-binary-string.py
class Solution:
    def minOperations(self, s: str) -> int:
        count = 0

        for i in range(1, len(s)):
            if s[i - 1] == s[i]:
                if s[i - 1] == '0':
                    s = s[:i] + '1' + s[i + 1:]
                    count += 1
                elif s[i - 1] == '1':

                    s = s[:i] + '0' + s[i + 1:]
                    count += 1
                

        return min(count, len(s) - count)

        